source/adios2/CMakeLists.txt

```diff
@@ -269,6 +269,10 @@ if(ADIOS2_HAVE_HDF5)
 
   foreach(lib IN LISTS adios2_hdf5_objs)
     target_link_libraries(${lib} PRIVATE ${HDF5_C_LIBRARIES})
+    message(STATUS HDF5_VERSION)
+    if(HDF5_VERSION VERSION_LESS_EQUAL 1.11.0)
+      target_compile_definitions(${lib} PRIVATE H5_USE_110_API)
+    endif()
```

Comment:
You should probably remove the status message, assuming it was for debugging.

But this is also not the correct way of handling this -- that macro is something that hdf5 1.12 understands -- if it is defined, it'll make the API compatible to 1.10. That is,
```
 if(HDF5_VERSION VERSION_GREATER_EQUAL 1.12.0)
      target_compile_definitions(${lib} PRIVATE H5_USE_110_API)
  endif()
```

by itself should fix the issue.

Reply:
if (version >= 1.12)
   define(H5_USE_110_API)? 

I think you may not mean it.

Reply:
I removed the status message.  Good catch, thanks!

I tried with both hdf5. 1.10 and 1.12. My original ifdef works.

Reply:
> if (version >= 1.12)
> define(H5_USE_110_API)?
> 
> I think you may not mean it.

Yes, I did mean it. H5_USE_110_API is a macro support by hdf5 1.12 that activates compatibility with 1.10, so then you don't have to change your code.

@chuckatkins ?